src/test/java/uk/gov/pay/directdebit/webhook/gocardless/resources/WebhookGoCardlessResourcePaymentActionsIT.java

```diff
@@ -0,0 +1,219 @@
+
+        Map<String, Object> payment = testContext.getDatabaseTestHelper().getPaymentById(paymentFixture.getId());
+        assertThat(payment.get("state"), is("INDEMNITY_CLAIM"));
+        assertThat(payment.get("state_details"), is("authorisation_disputed"));
+        assertThat(payment.get("state_details_description"), is("The customer has disputed that the amount taken differs from the amount they were " +
+                "notified of."));
+    }
+
+    @Test
+    @Ignore
+    public void chargebackCancelledChangesStateToPaidOut() throws IOException {
+        postWebhook("gocardless-webhook-payment-chargeback-cancelled.json",
+                "d87c8a2b43272c2175d4c50adc608f52a91d26f2be17f43a73b44d5eac65f835");
+
+        Map<String, Object> payment = testContext.getDatabaseTestHelper().getPaymentById(paymentFixture.getId());
+        assertThat(payment.get("state"), is("PAID_OUT"));
+        assertThat(payment.get("state_details"), is("payment_confirmed"));
+        assertThat(payment.get("state_details_description"), is("The chargeback for this payment was reversed"));
+    }
+
+    @Test
+    @Ignore
+    public void paidOutChangesStateToPaidOut() throws IOException {
+        postWebhook("gocardless-webhook-payment-paid-out.json",
+                "744d261836938a7e9f79d2157ccd3be06765f4d63cb605b5e3299a132d9ab7b7");
+
+        Map<String, Object> payment = testContext.getDatabaseTestHelper().getPaymentById(paymentFixture.getId());
+        assertThat(payment.get("state"), is("PAID_OUT"));
+        assertThat(payment.get("state_details"), is("payment_paid_out"));
+        assertThat(payment.get("state_details_description"), is("The payment has been paid out by GoCardless."));
+    }
+
+    @Test
+    @Ignore
+    public void chargebackSettledChangesStateToIndemnityClaim() throws IOException {
+        postWebhook("gocardless-webhook-payment-chargeback-settled.json",
+                "c663b23fb23b655b8181390235e015e2c7c50198403192505820d19e9af9ba24");
+
+        Map<String, Object> payment = testContext.getDatabaseTestHelper().getPaymentById(paymentFixture.getId());
+        assertThat(payment.get("state"), is("INDEMNITY_CLAIM"));
+        assertThat(payment.get("state_details"), is("chargeback_settled"));
+        assertThat(payment.get("state_details_description"), is("This charged back payment has been settled against a payout."));
+    }
+
+    private void postWebhook(String webhookBodyResourceName, String webhookSignature) throws IOException {
+        given().port(testContext.getPort())
+                .body(Resources.toString(Resources.getResource(webhookBodyResourceName), UTF_8))
+                .header("Webhook-Signature", webhookSignature)
+                .accept(APPLICATION_JSON)
+                .post("/v1/webhooks/gocardless")
+                .then()
+                .statusCode(OK.getStatusCode());
+    }
+
+    private void stubEmail(EmailTemplate template) throws JsonProcessingException {
```

Comment:
There's some duplication with the Mandate test but since it needs the wiremock rule and its pretty basic I like that it keeps the test easier to read by having it here also.